exec.go

```diff
@@ -80,38 +86,130 @@ func (rnr *execRunner) run(ctx context.Context, c *execCommand, s *step) error {
 
 		o.capturers.captureExecStdin(c.stdin)
 	}
-	cmd.Stdout = stdout
-	cmd.Stderr = stderr
+
+	stdout, err := cmd.StdoutPipe()
+
+	if err != nil {
+		return fmt.Errorf("error creating StdoutPipe: %w", err)
+	}
+
+	stderr, err := cmd.StderrPipe()
+
+	if err != nil {
+		return fmt.Errorf("error creating StderrPipe: %w", err)
+	}
+
+	err = cmd.Start()
+	if err != nil {
+		return fmt.Errorf("error starting command: %w", err)
+	}
+
+	done := make(chan execResult)
+	go func() {
+		var sob, seb bytes.Buffer
+		scanner := bufio.NewScanner(io.TeeReader(stdout, io.MultiWriter(&sob, io.Discard)))
+		o.capturers.captureExecStdoutStart(c.command)
+		for scanner.Scan() {
```

Comment:
If I output line by line, will it be broken if I use control characters?

Reply:
Not all patterns have been verified, though.
```
go run cmd/runn/main.go run --scopes run:exec  --debug testdata/book/exec.yml
Run "exec" on "Exec test".steps[0]
-----START COMMAND-----
echo hello world!!
-----END COMMAND-----
>>>>>START EXEC STDOUT (echo hello world!!)>>>>>
hello world!!
<<<<<END EXEC STDOUT (echo hello world!!)<<<<<
>>>>>START EXEC STDERR (echo hello world!!)>>>>>
<<<<<END EXEC STDERR (echo hello world!!)<<<<<
Run "test" on "Exec test".steps[0]

Run "exec" on "Exec test".steps[1]
-----START COMMAND-----
cat
-----END COMMAND-----
-----START STDIN-----
hello world!!

-----END STDIN-----
>>>>>START EXEC STDOUT (cat)>>>>>
hello world!!
<<<<<END EXEC STDOUT (cat)<<<<<
>>>>>START EXEC STDERR (cat)>>>>>
<<<<<END EXEC STDERR (cat)<<<<<
Run "test" on "Exec test".steps[1]

Run "exec" on "Exec test".steps[2]
-----START COMMAND-----
basename `echo $0`
-----END COMMAND-----
>>>>>START EXEC STDOUT (basename `echo $0`)>>>>>
bash
<<<<<END EXEC STDOUT (basename `echo $0`)<<<<<
>>>>>START EXEC STDERR (basename `echo $0`)>>>>>
<<<<<END EXEC STDERR (basename `echo $0`)<<<<<
Run "test" on "Exec test".steps[2]

Run "exec" on "Exec test".steps[3]
-----START COMMAND-----
for i in {1..5}
do
  echo -e "$i second\telapsed\07$i"
  sleep 1
done
-----END COMMAND-----
>>>>>START EXEC STDOUT (for i in {1..5}
do
  echo -e "$i second\telapsed\07$i"
  sleep 1
done)>>>>>
1 second        elapsed9
2 second        elapsed:
3 second        elapsed;
4 second        elapsed<
5 second        elapsed=
<<<<<END EXEC STDOUT (for i in {1..5}
do
  echo -e "$i second\telapsed\07$i"
  sleep 1
done)<<<<<
>>>>>START EXEC STDERR (for i in {1..5}
do
  echo -e "$i second\telapsed\07$i"
  sleep 1
done)>>>>>
<<<<<END EXEC STDERR (for i in {1..5}
do
  echo -e "$i second\telapsed\07$i"
  sleep 1
done)<<<<<
Run "test" on "Exec test".steps[3]

Run "exec" on "Exec test".steps[4]
-----START COMMAND-----
sleep 1000
-----END COMMAND-----
Run "test" on "Exec test".steps[4]
>>>>>START EXEC STDOUT (sleep 1000)>>>>>
.

1 scenario, 0 skipped, 0 failures
<<<<<END EXEC STDOUT (sleep 1000)<<<<<
```